src/librustc/middle/astencode.rs

```diff
@@ -144,7 +151,10 @@ pub fn decode_inlined_item<'tcx>(cdata: &cstore::crate_metadata,
             cdata: cdata,
             tcx: tcx,
             from_id_range: from_id_range,
-            to_id_range: to_id_range
+            to_id_range: to_id_range,
+            translate_spans: cdata.codemap_import_info.len() > 0 &&
```

Comment:
would it be harmful to always translate spans? I don't understand why the first predicate would be false - is this for backwards compatibility with existing crates? If we do the translate work and it is unnecessary (second predicate) - is that a real waste of resources? Does if actually introduce an error?

In general, I like to avoid flags like this - I feel the increased scope for error outweighs the performance benefit. But if it is necessary, then leave it.